src/StatusQ/Controls/StatusBaseButton.qml

```diff
@@ -19,131 +20,110 @@ Rectangle {
-    }
+    property StatusAssetSettings asset: StatusAssetSettings { }
 
     property bool loading: false
 
-    property alias hovered: sensor.containsMouse
+    property color normalColor
+    property color hoverColor
+    property color disabledColor
+
+    property color textColor
+    property color disabledTextColor
+    property color borderColor: "transparent"
 
     property int size: StatusBaseButton.Size.Large
     property int type: StatusBaseButton.Type.Normal
 
-    property alias text: label.text
-    property alias font: label.font
-
-    property real defaultLeftPadding: size === StatusBaseButton.Size.Large ? 24 : 12
-    property real defaultRightPadding: size === StatusBaseButton.Size.Large ? 24 : 12
-    property real defaultTopPadding: {
-        switch (size) {
-            case StatusBaseButton.Size.Tiny:
-              return 5
-            case StatusBaseButton.Size.Small:
-              return 10
-            case StatusBaseButton.Size.Large:
-            default:
-              return 11
-        }
-    }
-    property real defaultBottomPadding: {
-        switch (size) {
+    property int defaultHorizontalPadding: (horizontalPadding > 0) ? horizontalPadding : ((size === StatusBaseButton.Size.Large) ? 24 : 12)
+    property int defaultVerticalPadding: {
+        if (verticalPadding > 0) {
+            verticalPadding;
+        } else {
+            switch (size) {
             case StatusBaseButton.Size.Tiny:
-              return 5
+                return 5;
             case StatusBaseButton.Size.Small:
-              return 10
+                return 10;
             case StatusBaseButton.Size.Large:
             default:
-              return 11
+                return 11;
+            }
         }
     }
 
-
-    property real leftPadding: defaultLeftPadding
-    property real rightPadding: defaultRightPadding
-    property real topPadding: defaultTopPadding
-    property real bottomPadding: defaultBottomPadding
-
-    property color normalColor
-    property color hoverColor
-    property color disaledColor
-
-    property color textColor
-    property color disabledTextColor
-
-    signal pressed(var mouse)
-    signal released(var mouse)
-    signal clicked(var mouse)
-    signal pressAndHold(var mouse)
-
-    property bool highlighted: false
-
-
-    /// Implementation
-    implicitWidth: layout.width + statusBaseButton.leftPadding + statusBaseButton.rightPadding
-    implicitHeight: layout.height + statusBaseButton.topPadding + statusBaseButton.bottomPadding
-
-
-    radius: size !== StatusBaseButton.Size.Tiny ? 8 : 6
-
-    color: {
-        if (statusBaseButton.enabled)
-            return sensor.containsMouse || highlighted ? hoverColor
-                                                       : normalColor;
-        return disaledColor
+    implicitWidth: (contentItem.width + root.leftPadding + root.rightPadding)
+    implicitHeight: (contentItem.height + root.topPadding + root.bottomPadding)
```

Comment:
You don't need to specify these

Reply:
how come?

Reply:
They are calculated automatically from the contentItem/padding, try it :)